### ISSUE TITLE
Crash when adding new podcast by name

### ISSUE THREAD

**Issue by erazemk** [y20k/escapepod#65]
**Describe the bug**
Then entering a podcast name into the "Find Podcast" field, the app crashes after a few seconds (either when pressing 'Add' or when not).

This is the crash report I get:
```
time: 1616145188888
msg: java.lang.NullPointerException: Parameter specified as non-null is null: method k.b.a.b.a.v, parameter $this$collectionSizeOrDefault
stacktrace: java.lang.NullPointerException: Parameter specified as non-null is null: method k.b.a.b.a.v, parameter $this$collectionSizeOrDefault
	at k.b.a.b.a.v(Unknown Source:2)
	at b.a.a.d.d.e(SourceFile:7)
	at b.a.a.d.d$d.a(SourceFile:3)
	at k.a.b.x.i.b(SourceFile:4)
	at k.a.b.g$b.run(SourceFile:4)
	at android.os.Handler.handleCallback(Handler.java:883)
	at android.os.Handler.dispatchMessage(Handler.java:100)
	at android.os.Looper.loop(Looper.java:214)
	at android.app.ActivityThread.main(ActivityThread.java:7356)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:491)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:925)
```

**To Reproduce**
1. Open app
2. Press "Add new podcast"
3. Start entering text (not pasting link)
4. App crashes

**Expected behavior**
App finds the wanted podcast.

**Device:**
 - Device: Xiaomi Redmi Note 3 Pro (kate)
 - OS: HavocOS
 - App version: 1.0.1 (downloaded from Github)

**Additional context:**
The crash doesn't occur on the latest F-Droid version (0.9.5).

**Comment by y20k** [comment]
@erazemk 
Thanks for reporting this bug. Having the search not work is pretty significant!! I cannot reproduce the problem. :-(

Can you please check which search provider causes the problem (selectable via Settings)? Or do both providers crash the app?

**Comment by erazemk** [comment]
The problem seems to only occur with Podcastindex.org

**Comment by y20k** [comment]
Okay. Thanks! Any Non-Latin characters in your query? I am left to guessing at the moment, sorry.

**Comment by erazemk** [comment]
>  Any Non-Latin characters in your query?

Nope, I tried searching "Linux" as in "Linux Action News", but it crashed.
If I can help you debug this by providing more info (a full logcat maybe), I'd be happy to do so, just say what you need.

**Comment by y20k** [comment]
You could try to recreate the crash with this version of the app
* [escapepod-app-testing-1.0.1-001.apk](https://drive.google.com/file/d/1laIaRz1mNGoLzuETZkqwUv5DWfx_haUg/view?usp=sharing)

This test-build has obfuscation disabled. The stack trace should be more informative.

**Comment by erazemk** [comment]
The crash doesn't seem to occur with the test build, the podcasts are found even when using Podcastindex.

**Comment by y20k** [comment]
I tested it the other way around. Installed the version from GitHub, in which the class names are obfuscated. I was able to reproduce the problem. Yay!! Obfuscation seems to be the issue. I will fix this later today and publish a new version. Thank you for reporting the problem.

**Comment by erazemk** [comment]
Thanks for fixing this so fast :)

**Comment by y20k** [comment]
I think I fixed the crash and I published a new version of the app ([v1.0.2](https://github.com/y20k/escapepod/releases/tag/v1.0.2)). Thank you for your quick responses and support.
